code/aggregation/aggregate-raw-data.R

```diff
@@ -345,25 +354,24 @@ rm(saphon.raw, saphon.ipa)
 ## combine into one data frame
 data.sources.list <- list(ph.data, aa.data, spa.data, upsid.data,
                           ra.data, gm.data, saphon.data)
-all.data <- do.call(rbind.fill, data.sources.list)
-all.data <- all.data[with(all.data, order(LanguageCode, Source)),]
-## remove all tiebars
+all.data <- do.call(rbind, data.sources.list)
+all.data <- all.data[with(all.data, order(LanguageCode, Source, InventoryID)),]
+## uniqueify inventory IDs across data sources
+all.data$InventoryID <- with(all.data, paste(LanguageCode, Source, InventoryID, sep="-"))
+## make sure all tiebars are removed (should already be cleaned from source)
 all.data$Phoneme <- stri_replace_all_fixed(all.data$Phoneme, replacement="", pattern="͡")
 all.data$Phoneme <- stri_replace_all_fixed(all.data$Phoneme, replacement="", pattern="͜")
 ## should all be denormalized already, but make sure
 all.data$Phoneme <- denorm(all.data$Phoneme)
 all.data$Allophones <- denorm(all.data$Allophones)
-## make boolean column "Marginal" and remove angle brackets
-all.data <- markMarginal(all.data)
```

Comment:
markMarginal now happens earlier (before different data sources are merged) so that sources that don't indicate marginal phonemes can get `NA` values instead of all `FALSE` values.